lib/vyasa_web/live/display_manager/display_live.ex

```diff
@@ -403,34 +411,53 @@ defmodule VyasaWeb.DisplayManager.DisplayLive do
   def handle_event(
         "createMark",
         %{"body" => body},
-        %{assigns: %{kv_verses: verses,
-                     marks: [%Mark{state: :draft, verse_id: v_id, binding: binding} = d_mark | marks],
-                     device_type: device_type}} = socket
+        %{
+          assigns: %{
+            kv_verses: verses,
+            marks: [%Mark{state: :draft, verse_id: v_id, binding: binding} = d_mark | marks],
+            device_type: device_type,
+            ui_state: ui_state
+          }
+        } = socket
       ) do
     {:noreply,
      socket
      |> assign(:marks, [%{d_mark | body: body, state: :live} | marks])
      |> stream_insert(
-       :verses, %{verses[v_id] | binding: binding}
+       :verses,
+       %{verses[v_id] | binding: binding}
      )
-     |> assign(:show_media_bridge?, should_show_media_bridge(device_type, false))}
+     |> assign(:ui_state, %UiState{
+       ui_state
+       | show_media_bridge?: should_show_media_bridge(device_type, false)
+     })}
   end
 
   # when user remains on the the same binding
   def handle_event(
         "createMark",
         %{"body" => body},
-        %{assigns: %{kv_verses: verses,
-                     marks: [%Mark{state: :live, verse_id: v_id, binding: binding} = d_mark | _] = marks,
-                     device_type: device_type}} = socket
+        %{
+          assigns: %{
+            kv_verses: verses,
+            marks: [%Mark{state: :live, verse_id: v_id, binding: binding} = d_mark | _] = marks,
+            device_type: device_type,
+            ui_state: ui_state
+          }
+        } = socket
       ) do
     {:noreply,
      socket
      |> assign(:marks, [%{d_mark | body: body, state: :live} | marks])
      |> stream_insert(
-       :verses, %{verses[v_id] | binding: binding}
+       :verses,
+       %{verses[v_id] | binding: binding}
      )
-     |> assign(:show_media_bridge?, should_show_media_bridge(device_type, false))}
+     |> assign(:show_media_bridge?, should_show_media_bridge(device_type, false))
+     |> assign(:ui_state, %UiState{
+       ui_state
+       | show_media_bridge?: should_show_media_bridge(device_type, false)
```

Comment:
should be bundled as helper fns with names that are expressive such as UIState.unhide_media()

They can take in socket as variable where UIState struct is defined allowing us to seperate code that manipulates the DOM with relation to View related concerns (user is on mobile in keyboard state, wide desktop view) vs the more classic DOM events that are triggered of behavioural causes

Reply:
nice suggestion, updated it